src/Application/Models/Bank/Asset.js

```diff
@@ -39,6 +40,14 @@ class Asset {
     this._multiplier += additionalPercentage;
   }
 
+  unlock() {
+    this._isUnlocked = true;
+  }
```

Comment:
Immutable just means that whenever you want to change an Asset, remove that Asset and add a brand new one that looks a lot like like the old Asset, but with the things you want changed changed.  Mutability is baked pretty hard into `Asset` so the path to immutability is not 100% simple.  I'd say it'd be better to take the things that change and make them part of a flux store, since that's where state that changes belongs